packages/block-editor/src/components/block-draggable/use-scroll-when-dragging.js

```diff
@@ -60,15 +60,42 @@ export default function useScrollWhenDragging( dragElement ) {
 	);
 
 	const scrollOnDragOver = useCallback( ( event ) => {
-		const distanceY = event.clientY - dragStartY.current;
-		if ( distanceY > SCROLL_INACTIVE_DISTANCE_PX ) {
-			velocityY.current =
-				VELOCITY_MULTIPLIER *
-				( distanceY - SCROLL_INACTIVE_DISTANCE_PX );
-		} else if ( distanceY < -SCROLL_INACTIVE_DISTANCE_PX ) {
-			velocityY.current =
-				VELOCITY_MULTIPLIER *
-				( distanceY + SCROLL_INACTIVE_DISTANCE_PX );
+		const scrollParentHeight = scrollParentY.current.offsetHeight;
+		const offsetDragStartPosition =
+			dragStartY.current - scrollParentY.current.offsetTop;
+		const offsetDragPosition =
+			event.clientY - scrollParentY.current.offsetTop;
+
+		if ( event.clientY > offsetDragStartPosition ) {
+			// User is dragging downwards.
+			const moveableDistance = Math.max(
+				scrollParentHeight -
+					offsetDragStartPosition -
+					SCROLL_INACTIVE_DISTANCE_PX,
+				0
+			);
+			const dragDistance = Math.max(
+				offsetDragPosition -
+					offsetDragStartPosition -
+					SCROLL_INACTIVE_DISTANCE_PX,
+				0
+			);
+			const distancePercentage = dragDistance / moveableDistance;
+			velocityY.current = VELOCITY_MULTIPLIER * distancePercentage;
+		} else if ( event.clientY < offsetDragStartPosition ) {
+			// User is dragging upwards.
+			const moveableDistance = Math.max(
+				offsetDragStartPosition - SCROLL_INACTIVE_DISTANCE_PX,
+				0
+			);
+			const dragDistance = Math.max(
+				offsetDragStartPosition -
+					offsetDragPosition -
+					SCROLL_INACTIVE_DISTANCE_PX,
+				0
+			);
+			const distancePercentage = dragDistance / moveableDistance;
```

Comment:
These lines appear to be duplicates of lines in the previous if-clause - is there a way to eliminate this duplication?